### PR TITLE
Add behat functional test case for push command

### PR DESCRIPTION
**Pull request by olaurendeau**
Trying to add some more unit tests, I realized openl10n-cli code isn't that simple and wouldn't be totally trivial to unit test. So I thought that some behat test cases (strongly inspired by the https://github.com/Behat/Behat feature suite herself) could quickly protect the project from regression.

Pending for https://github.com/openl10n/openl10n-sdk/pull/13 to build properly on `php >= 5.5`

Guzzle mocks weren't available on Guzzle 4, so I had to upgrade https://github.com/openl10n/openl10n-sdk on `dev-master`, may could you release a new version of https://github.com/openl10n/openl10n-sdk ?

Of course if adding functionnal tests wasn't in your plans feel free to close this pull request.

### CODE REVIEW

**Comment by olaurendeau** [comment]
Building noticed me that using Guzzle 6.0 in https://github.com/openl10n/openl10n-sdk will remove php 5.4 support, is it fine for you ?

**Comment by olaurendeau** [comment]
up

File: composer.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
     "type": "project",
     "require": {
         "php": "^5.4",
-        "openl10n/sdk": "^0.1",
+        "openl10n/sdk": "dev-master@dev",
```
Comment (MattKetmo) [inline composer.json]: You can use now `^0.2`

File: .travis.yml [hunk 2]
```diff
@@ -2,7 +2,10 @@ language: php
 php:
   - 5.6
   - 5.5
-  - 5.4
 before_script:
+  - composer selfupdate
   - composer install
+  - export PATH=./bin:$PATH
```
Comment (MattKetmo) [inline .travis.yml]: Is this new PATH useful?

File: composer.json [hunk 3]
```diff
@@ -23,7 +24,8 @@
     },
     "scripts": {
         "test": [
-            "phpunit --colors=always"
+            "phpunit --colors=always",
+            "behat --format=progress"
```
Comment (MattKetmo) [inline composer.json]: you can add the `--colors` option for behat

**Comment by olaurendeau** [comment]
@MattKetmo Are you sure about the php 5.4 support removal ? https://github.com/olaurendeau/openl10n-cli/commit/9789319d73b2ae3e07726b436bc7dfde2ae5db43

**Comment by MattKetmo** [comment]
Since openl10n/sdk requires guzzle 6 which requires php55+ yes there is no choice but removing the php54 compat. Btw the composer.json should be modified too to explicitly require `"php": "^5.5"`.

**Comment by MattKetmo** [comment]
Thanks for the work ;)

**Pull request merged**